awx/main/scheduler/task_manager.py

```diff
@@ -302,14 +304,10 @@ def create_inventory_update(self, task, inventory_source_task):
         # self.process_inventory_sources(inventory_sources)
         return inventory_task
 
-    def capture_chain_failure_dependencies(self, task, dependencies):
+    def add_dependencies(self, task, dependencies):
         with disable_activity_stream():
             task.dependent_jobs.add(*dependencies)
 
-            for dep in dependencies:
-                # Add task + all deps except self
-                dep.dependent_jobs.add(*([task] + [d for d in dependencies if d != dep]))
```

Comment:
Probably related to the migration, but could you explain how you're able to remove this?

Reply:
yeah this change here _could_ have implications

consider this setup (JobA and JobB both share the projUpdate dependency. But only JobA depends on invUpdate)

```
            JobA                    JobB
            |  |                      |
            |  |                      |
            |  |                      |
            |  |                      |
invUpdate <-+  +--->  projUpdate <----+
````

Currently on devel, invUpdate and projUpdate are linked. If one fails, it brings down the other. If invUpdate fails, it causes projUpdate to fail, and thus JobB would fail, even though JobB never depended on invUpdate in the first place.

I can't think of why this behavior is needed.

Because of the nature of sharing dependencies, if a project update or inventory update fails, it should only fail the jobs that directly depend on it, not other stuff.

Reply:
the dependent_jobs field plays a role in determining which jobs fail, as I mention above.

The other role for dependent_jobs is to determine whether a job is blocked from running.

in task_manager.py
```python
if not task.dependent_jobs_finished():
    blocked_by = task.dependent_jobs.first()
```
On devel this really only comes into play for Jobs

type Job
```python
def dependent_jobs_finished(self):
    for j in self.dependent_jobs.all():
        if j.status in ['pending', 'waiting', 'running']:
            return False
    return True
```

But for ProjectUpdate and InventoryUpdate, this always returned True
```python
def dependent_jobs_finished(self):
    return True
```

Therefore, the removed code in the diff doesn't really have an effect here. Jobs still have the same dependent_jobs as before. The only difference is that InventoryUpdate and ProjectUpdate do not have each other as dependencies, but this doesn't affect the `is_blocked` behavior.